covid_graphs/covid_graphs/formula.py

```diff
@@ -1,81 +1,77 @@
+        self.end_date = end_date
+        self.label = label
 
-def ATG_formula(TG, A, exponent, min_case_count=200):
-    text = r"$\frac{_A}{_TG} \cdot \left(\frac{t}{_TG}\right)^{_expon} / e^{t/_TG}$"
-    text = text.replace("_A", f"{A:.0f}").replace("_TG", f"{TG}").replace("_expon", f"{exponent}")
-    # Day of the second inflection point
-    second_ip_day = math.ceil(TG * (exponent + math.sqrt(exponent)))
-    return Formula(
-        lambda t: (A / TG) * (t / TG) ** exponent / np.exp(t / TG),
-        text,
-        second_ip_day,
-        min_case_count,
-    )
+        self.xs = [
+            self.start_date + datetime.timedelta(days=d)
+            for d in range((self.end_date - self.start_date).days)
+        ]
+        self.ys = [func(x) for x in self.xs]
 
+    def get_maximum(self) -> Tuple[datetime.date, float]:
+        """
+        Return TODO
+        """
+        idx = np.argmax(self.ys)
+        return self.xs[idx], self.ys[idx]
 
-def first_day_of_curve(cumulative_active, formula):
-    """
-    Returns the first day of the curve formed by 'formula'.
 
-    It's the first day for which the number of cumulative active is more than
-    formula.min_case_count.
-    """
-    return np.argmax(cumulative_active > formula.min_case_count)
+class CurveConstructor:
+    @abstractmethod
+    def get_curve(self, country_report: CountryReport) -> Curve:
+        pass
 
 
-class Curve:
-    """
-    Class containing a curve derived from a formula.
+@dataclass
+class AtgCurveConstructor(CurveConstructor):
+    tg: float
+    a: float
+    exponent: float
+    min_case_count: int
 
-    cumulative_active - list of cumulative active cases
-    first_idx - 0-based index where the graph starts, so from cumulative_active[first_idx:].
-    last_idx - 0-based index where the graph ends
-    first_date - date corresponding to cumulative_active[0]
-    """
+    def get_curve(self, country_report: CountryReport) -> Curve:
+        label = r"$\frac{_A}{_TG} \cdot \left(\frac{t}{_TG}\right)^{_expon} / e^{t/_TG}$"
+        label = (
+            label.replace("_A", f"{self.a:.0f}")
+            .replace("_TG", f"{self.tg}")
+            .replace("_expon", f"{self.exponent}")
+        )
 
-    def __init__(self, formula, cumulative_active, first_idx, last_idx, first_date):
-        self.text = formula.text
-        start_idx = first_day_of_curve(cumulative_active, formula)
-        length = last_idx - start_idx + 1
-        self.y = formula.lambd(np.arange(length) + 1)
-        self.t = np.arange(length) + 1 + (start_idx - first_idx)
-        self.date_list = [
-            (first_date + timedelta(days=d)).strftime("%Y-%m-%d")
-            for d in range(start_idx, last_idx + 1)
-        ]
+        # Display values from the first day for which the number of cumulative active is
+        # more than min_case_count. This day is also "day one".
+        start_idx = np.argmax(country_report.cumulative_active >= self.min_case_count)
```

Comment:
Katka and Rišo use a strict inequality, I think that's what's causing the shift for you. This is what we had before
```
return np.argmax(cumulative_active > formula.min_case_count)
```

Strict inequality is weird though, I'd suggest updating the predictions instead by bumping the minimum by +1.

Reply:
Used `>=` according to the new specification. I understood that `min_cases` define `t=1` day. It looks like it works ok now, only `Austria` was adjusted manually in `predictions.py`.